src/parser/shaman/elemental/modules/talents/Ascendance.js

```diff
@@ -36,6 +55,30 @@ class Ascendance extends Analyzer {
   }
 
   on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    const target = this.enemies.getEntity(event);
+
+
+    if(spellId === SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id) {
+      if(target && target.getBuffUptime(SPELLS.FLAME_SHOCK.id) < ASCENDENCE_DURATION){
```

Comment:
Ascendance cast doesn't have a target, so this will never be true. How about checking on the first cast with a target after ascendance instead?